sequencer/src/lib.rs

```diff
@@ -589,23 +589,24 @@ pub mod testing {
         ExecutionType, HotShotConfig, PeerConfig,
     };
     use portpicker::pick_unused_port;
+    use std::collections::HashMap;
     use std::time::Duration;
     use vbs::version::Version;
 
     const STAKE_TABLE_CAPACITY_FOR_TEST: u64 = 10;
 
     pub async fn run_test_builder<const NUM_NODES: usize>(
+        mut url: Url,
         port: Option<u16>,
-    ) -> (Box<dyn BuilderTask<SeqTypes>>, Url) {
-        let builder_config = if let Some(port) = port {
-            SimpleBuilderConfig { port }
-        } else {
-            SimpleBuilderConfig::default()
-        };
+    ) -> Box<dyn BuilderTask<SeqTypes>> {
+        let port = port.unwrap_or_else(|| pick_unused_port().expect("No available ports"));
+        url.set_port(Some(port)).expect("Failed to set port");
```

Comment:
Unclear why we're taking a URL just to change the port, vs just `localhost:{port}`

Reply:
Good point.. honestly not sure why that was there.